guides/source/configuring.md

```diff
@@ -383,6 +383,12 @@ All these configuration options are delegated to the `I18n` library.
   having to send a query to the database to get this information.
   Defaults to `true`.
 
+* `config.active_record.collection_cache_versioning` enables the same cache key
+  to be reused when the object being cached of type `ActiveRecord::Relation`
+  changes by moving the volatile information (max updated at and count) of
+  the relation's cache key into the cache version to support recycling cache key.
+  Defaults to `true`.
```

Comment:
We should talk about the default value in the context of AR when no Railtie code is applied, so it should be `false`: https://github.com/rails/rails/blob/69fcaaaf15880dc20b9ec19a9b839f375a64a1d2/activerecord/lib/active_record/integration.rb#L32

See similar cases https://github.com/rails/rails/pull/35873#issuecomment-480333333, https://github.com/rails/rails/pull/34855#issuecomment-451287368

https://github.com/rails/rails/commit/dc17bc0a1017ca99692da6ec3d1cf0889355c767 added to https://edgeguides.rubyonrails.org/configuring.html#with-6-0 `config.active_record.collection_cache_versioning: true` to tell readers of the guide that it is `true` when  `load_defaults('6.0')` is applied.

Reply:
Thanks, @bogdanvlviv. I will review the behavior correctly and send a new PR to fix the docs accordingly.

Reply:
Thanks, @bogdanvlviv. I will review the behavior correctly and send a new PR to fix the docs accordingly.

Reply:
@bogdanvlviv When `config.load_defaults 6.0` is added to `config/application.rb` in my demo app, I get `true` value for `config.active_record.collection_cache_versioning` config.

```
$ rails c
Loading development environment (Rails 6.0.0.rc1)
>> Rails.application.config.active_record.collection_cache_versioning
=> true
```

Whereas, when `config.load_defaults 6.0` is not added to `config/application.rb`, I get `nil` for the same configuration.

```
$ rails c
Loading development environment (Rails 6.0.0.rc1)
>> Rails.application.config.active_record.collection_cache_versioning
=> nil
```

Do you think this is the correct behavior? Shouldn't it be `false` or is it being consumed using `delete(:collection_cache_versioning)` somewhere?

Reply:
> Do you think this is the correct behavior? Shouldn't it be false or is it being consumed using delete(:collection_cache_versioning) somewhere?

Yes, the behavior is correct.

You should check for value of `ActiveRecord::Base.collection_cache_versioning` instead of `config.active_record.collection_cache_versioning`

Reply:
@bogdanvlviv Thank you for the information. I checked the value of `ActiveRecord::Base.collection_cache_versioning` and it appears correctly the way you described it.